stan/math/rev/core/vari.hpp

```diff
@@ -200,7 +209,8 @@ class vari_value<T, require_eigen_dense_base_t<T>> : public vari_base {
   using eigen_map = Eigen::Map<PlainObject, Eigen::Aligned8>;
   using vari_type = vari_value<T, require_eigen_dense_base_t<T>>;
   eigen_scalar* val_mem_;  // Pointer to memory allocated on the stack for val_
-  eigen_scalar* adj_mem_;  // Pointer to memory allocated on the stack for adj_
+  mutable eigen_scalar*
+      adj_mem_;  // Pointer to memory allocated on the stack for adj_
```

Comment:
Do we ever need to change a value of a vari?

Reply:
Good catch! No idt we will ever need to change the pointer directly here just the underlying values